pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -80,6 +86,9 @@ type NodeController struct {
 	recorder           record.EventRecorder
 	registerRetryCount int
 	podEvictionTimeout time.Duration
+	// per Node map storing last observed Status togheter with a local time when it was observed.
+	// This timestamp is to be used instead of LastProbeTime stored in Condition.
```

Comment:
I would add a reason why this timestamp is used, e.g. to avoid cluster time skew